gcloud-java-compute/src/main/java/com/google/gcloud/compute/StandardDiskConfiguration.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import com.google.api.services.compute.model.Disk;
+
+import java.util.Objects;
+
+/**
+ * A Google Compute Engine standard persistent disk configuration. Allows to create a disk given its
+ * type and size.
+ *
+ * @see <a href="https://cloud.google.com/compute/docs/disks/">Block Storage</a>
+ */
+public class StandardDiskConfiguration extends DiskConfiguration {
+
+  private static final long serialVersionUID = -6974045909359567054L;
+
+  /**
+   * A builder for {@code StandardDiskConfiguration} objects.
+   */
+  public static class Builder
+      extends DiskConfiguration.Builder<StandardDiskConfiguration, Builder> {
+
+    private Builder() {
+      super(Type.STANDARD);
+    }
+
+    private Builder(StandardDiskConfiguration diskInfo) {
+      super(Type.STANDARD, diskInfo);
+    }
+
+    private Builder(Disk diskPb) {
+      super(Type.STANDARD, diskPb);
+    }
+
+    /**
+     * Creates a {@code StandardDiskConfiguration} object.
+     */
+    @Override
+    public StandardDiskConfiguration build() {
+      return new StandardDiskConfiguration(this);
+    }
+  }
+
+  private StandardDiskConfiguration(StandardDiskConfiguration.Builder builder) {
+    super(builder);
+  }
+
+  @Override
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  @Override
+  public final int hashCode() {
+    return Objects.hash(baseHashCode());
+  }
+
+  @Override
+  public final boolean equals(Object obj) {
+    return obj instanceof StandardDiskConfiguration && baseEquals((StandardDiskConfiguration) obj);
+  }
+
+  @Override
+  StandardDiskConfiguration setProjectId(String projectId) {
+    if (diskType() == null || diskType().project() != null) {
+      return this;
```

Comment:
In the other disk configuration subclasses, you use the project ID given as a parameter, no matter what (probably because there are two IDs that need to be updated with a project ID, and it's cleaner and simpler to replace both rather than check for an existing project ID).  Do you think there's any issue that the behavior here is slightly different (we only set the project ID if one isn't set already)?